### PR TITLE
Add table aws_ec2_settings. closes #386

### LINKED ISSUE #386: Add table aws_ec2_settings
**References**
https://github.com/turbot/steampipe-plugin-aws/pull/352#issuecomment-837251949

### PR DESCRIPTION
**Pull request by pskrbasu**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/aws_ec2_regional_settings []

PRETEST: tests/aws_ec2_regional_settings

TEST: tests/aws_ec2_regional_settings
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_ebs_encryption_by_default.named_test_resource: Creating...
aws_kms_key.named_test_resource: Creating...
aws_ebs_encryption_by_default.named_test_resource: Creation complete after 2s [id=terraform-20210609090133143400000001]
aws_kms_key.named_test_resource: Creation complete after 6s [id=7294e615-bbcf-419d-bdad-03691d53096b]
aws_ebs_default_kms_key.named_test_resource: Creating...
aws_ebs_default_kms_key.named_test_resource: Creation complete after 3s [id=arn:aws:kms:us-east-1:123456789012:key/7294e615-bbcf-419d-bdad-03691d53096b]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

aws_account = 123456789012
aws_partition = aws
aws_region = us-east-1
resource_aka = arn:aws:kms:us-east-1:123456789012:key/7294e615-bbcf-419d-bdad-03691d53096b

Running SQL query: test-get-query.sql
[
  {
    "default_ebs_encryption_enabled": true,
    "default_ebs_encryption_key": "arn:aws:kms:us-east-1:123456789012:key/7294e615-bbcf-419d-bdad-03691d53096b",
    "region": "us-east-1",
    "title": "us-east-1 EC2 Settings"
  }
]
✔ PASSED

POSTTEST: tests/aws_ec2_regional_settings

TEARDOWN: tests/aws_ec2_regional_settings

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  default_ebs_encryption_enabled,
  default_ebs_encryption_key,
  title,
  region
from
  aws_ec2_settings;
+--------------------------------+------------------------------------------------------------------------------+--------------+----------------+
| default_ebs_encryption_enabled | default_ebs_encryption_key                                                   | title        | region         |
+--------------------------------+------------------------------------------------------------------------------+--------------+----------------+
| false                          | alias/aws/ebs                                                                | EC2 Settings | af-south-1     |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-south-1     |
| false                          | alias/aws/ebs                                                                | EC2 Settings | me-south-1     |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-east-1      |
| true                           | arn:aws:kms:ap-south-1:453319552164:key/24213245-62a4-40de-8790-2463a8e9550b | EC2 Settings | ap-south-1     |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-southeast-1 |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-central-1   |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-west-3      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-northeast-3 |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-west-2      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-northeast-2 |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-north-1     |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-northeast-1 |
| false                          | arn:aws:kms:us-east-2:453319552164:key/8dddf7e5-dd9c-4ae4-91c9-2aa928cce178  | EC2 Settings | us-east-2      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ca-central-1   |
| false                          | alias/aws/ebs                                                                | EC2 Settings | us-west-1      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | ap-southeast-2 |
| false                          | alias/aws/ebs                                                                | EC2 Settings | us-west-2      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | eu-west-1      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | us-east-1      |
| false                          | alias/aws/ebs                                                                | EC2 Settings | sa-east-1      |
+--------------------------------+------------------------------------------------------------------------------+--------------+----------------+
> select
  default_ebs_encryption_enabled,
  default_ebs_encryption_key,
  title,
  region
from
  aws_ec2_settings
where
  region='ap-south-1';
+--------------------------------+------------------------------------------------------------------------------+--------------+------------+
| default_ebs_encryption_enabled | default_ebs_encryption_key                                                   | title        | region     |
+--------------------------------+------------------------------------------------------------------------------+--------------+------------+
| true                           | arn:aws:kms:ap-south-1:453319552164:key/24213245-62a4-40de-8790-2463a8e9550b | EC2 Settings | ap-south-1 |
+--------------------------------+------------------------------------------------------------------------------+--------------+------------+
> select
  region,
  default_ebs_encryption_enabled,
  default_ebs_encryption_key
from
  aws_ec2_settings
where
  default_ebs_encryption_enabled='true';
+------------+--------------------------------+------------------------------------------------------------------------------+
| region     | default_ebs_encryption_enabled | default_ebs_encryption_key                                                   |
+------------+--------------------------------+------------------------------------------------------------------------------+
| ap-south-1 | true                           | arn:aws:kms:ap-south-1:453319552164:key/24213245-62a4-40de-8790-2463a8e9550b |
+------------+--------------------------------+------------------------------------------------------------------------------+
```
</details>

### CODE REVIEW

File: docs/tables/aws_ec2_settings.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+```sql
+select
+  default_ebs_encryption_enabled,
+  default_ebs_encryption_key,
+  title,
+  region
+from
+  aws_ec2_settings
+where
+  region='ap-south-1';
```
Comment (Subhajit97) [inline docs/tables/aws_ec2_settings.md:31]:
```suggestion
  region = 'ap-south-1';
```

File: docs/tables/aws_ec2_settings.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+```sql
+select
+  region,
+  default_ebs_encryption_enabled,
+  default_ebs_encryption_key
+from
+  aws_ec2_settings
+where
+  default_ebs_encryption_enabled='true';
```
Comment (Subhajit97) [inline docs/tables/aws_ec2_settings.md:45]:
```suggestion
  default_ebs_encryption_enabled;
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

File: aws/table_aws_ec2_settings.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+				Description: "The Amazon Resource Name (ARN) or alias of the default CMK for encryption by default.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getDefaultEBSVolumeEncryptionKey,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("EC2 Settings"),
```
Comment (cbruno10) [inline aws/table_aws_ec2_settings.go:42]: To better differentiate which row is from which region, are we able to change this to `<region> EC2 Settings`?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comments

**Comment by Subhajit97** [comment]
+--------------------------------+-----------------------------------------------------------------------------+-------------------------+-----------+------------+--------------+
| default_ebs_encryption_enabled | default_ebs_encryption_key                                                  | title                   | partition | region     | account_id   |
+--------------------------------+-----------------------------------------------------------------------------+-------------------------+-----------+------------+--------------+
| false                          | alias/aws/ebs                                                               | us-east-2 EC2 Settings  | aws       | us-east-2  | 123456789012 |
| false                          | alias/aws/ebs                                                               | af-south-1 EC2 Settings | aws       | af-south-1 | 123456789012 |
| false                          | arn:aws:kms:us-east-1:123456789012:key/47dc93a0-ca41-46b2-96bd-08980fa0ed7c | us-east-1 EC2 Settings  | aws       | us-east-1  | 123456789012 |
+--------------------------------+-----------------------------------------------------------------------------+-------------------------+-----------+------------+--------------+

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Some filenames still have the old name (`aws_ec2_settings` instead of `aws_ec2_regional_settings`), can you please update those?

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**